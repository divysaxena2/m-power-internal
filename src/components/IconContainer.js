import React, { Component } from 'react';
import getTheme from '../../native-base-theme/components'
import material from '../../native-base-theme/variables/material';
import { StyleSheet, View, Image } from 'react-native';
import { Button, StyleProvider, Icon, Footer, Text, FooterTab } from 'native-base';

type Props = {};
export default class IconContainer extends Component<Props> {
    render() {
        return (
            <StyleProvider style={getTheme(material)}>
                <View style={styles.container} >
                    <View style={{ paddingLeft: 10 }}>
                        <Image source={require('../assets/ribbon.png')} style={styles.img} />
                    </View>
                    <View>
                        <Text style={styles.iconTitle}>{this.props.text}</Text>
                    </View>
                </View>
            </StyleProvider >)
    }
}
const styles = StyleSheet.create({
    container: {
        height: 110,
        width: 80,
        paddingLeft: 20
    },
    iconTitle: {
        color: '#000000',
        //font- family: Roboto;
        fontSize: 13,
        letterSpacing: 0.62,
        lineHeight: 18,
        textAlign: 'center'
    },
    img: {
        height: 40,
        width: 34
    }

});