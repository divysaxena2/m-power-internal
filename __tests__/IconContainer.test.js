import 'react-native';
import React from 'react';
import IconContainer from '../src/components/IconContainer';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <IconContainer />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});