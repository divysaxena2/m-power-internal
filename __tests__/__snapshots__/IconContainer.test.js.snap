// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingLeft": 25,
    }
  }
>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 40,
          "width": 34,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Roboto",
          "fontSize": 16,
        },
        Object {
          "fontSize": 14,
          "height": 41,
          "lineHeight": 19,
          "textAlign": "center",
          "width": 60.2,
        },
      ]
    }
    uppercase={false}
  />
</View>
`;
